### PR TITLE
start work on a testing plan for devtools

### PR DESCRIPTION
**Pull request by devoncarew**
A rough draft of a testing plan for this project. I'd expect to use `package:test` to write tests, and each test would have the setup described in the markdown file. Comments welcome, and I'll work on adding an actual test as part of this PR.

@jacob314 @dantup

### CODE REVIEW

**Comment by DanTup** [comment]
Plan LGTM (though sounds like you're not ready to merge if you're going to add a test).

**Comment by devoncarew** [comment]
Yup, I'd like to try and get one test in, then iterate on that - add one or two more, and make sure that the common parts of the tests are abstracted out a bit (build the framework once we have a better idea of what the tests will look like).

**Comment by devoncarew** [comment]
OK, this PR now contains the integration testing framework and the first two integration tests.

test 1: start the devtools app and verify that you can switch pages
test 2: start the devtools app; start a fixture app; connect to the fixture app from devtools; switch to the 'logs' page in devtools; clear the log data; cause the fixture app to emit a message which should show up in the log view; verify that the devtools log page does show the log message

Ready for review; comments welcome. After landing, I'll follow up with a PR that runs the tests on travis (they just run via `pub run test`, but do need to be able to locate a copy of chrome, and need either a headed environment, or for chrome to run headlessly).

File: test/fixtures/logging_app.dart [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import 'dart:async';
+import 'dart:developer';
+import 'dart:io';
+
+// Allow a test driver to communicate with this app through the controller.
+final Controller controller = new Controller();
+
+void main() {
+  log('starting app');
+  print('starting app');
```
Comment (jacob314) [inline test/fixtures/logging_app.dart]: can the print calls be removed?

File: test/src/chrome.dart [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+
+  Chrome._(this.executable);
+
+  static Chrome locate() {
+    if (Platform.isMacOS) {
+      // /Applications/Google\ Chrome\ Canary.app/
+
+      //const String defaultPath = '/Applications/Google Chrome.app';
+      const String defaultPath = '/Applications/Google Chrome Canary.app';
+      //const String bundlePath = 'Contents/MacOS/Google Chrome';
```
Comment (jacob314) [inline test/src/chrome.dart]: I assume you meant to leave it at Chrome not Canary before checkin?

File: test/src/chrome.dart [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+      _exceptionThrownController.stream;
+
+  Future<Null> reload() => _wip.page.reload();
+
+  Future<dynamic> navigate(String url) => _wip.page.navigate(url);
+
+  WipConnection get wipConnection => _wip;
+}
+
+// TODO(devoncarew): I'm not sure this class is necessary.
```
Comment (jacob314) [inline test/src/chrome.dart]: probably remove this class.

File: test/fixtures/logging_app.dart (hunk 1 above)
Comment (devoncarew) [reply test/fixtures/logging_app.dart]: I'd like to keep them just for the sake of making it easier to debug this app in some circumstances.

File: test/src/chrome.dart (hunk 2 above)
Comment (devoncarew) [reply test/src/chrome.dart]: Yup! Will update.

File: test/src/chrome.dart (hunk 3 above)
Comment (devoncarew) [reply test/src/chrome.dart]: will do

**Pull request merged**